cpu/atmega2560/periph/gpio.c

```diff
@@ -121,8 +139,6 @@ int gpio_init_int(gpio_t pin, gpio_pp_t pullup, gpio_flank_t flank,
         return -1;
     }
 
-    gpio_set(pin);
```

Comment:
Yes. gpio_set will set the bit in PORTx aswell. enabling the pullup no matter  what